src/operator/tensor/la_op.h

```diff
@@ -428,7 +429,14 @@ inline bool DetShape(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(in[ndim-2], in[ndim-1]) << "Input A's last two dimension must be equal";
   mxnet::TShape out;
   if (ndim == 2) {
-    out = mxnet::TShape(1, 1);
+    if (Imperative::Get()->is_np_shape()) {
+      out = mxnet::TShape(0, 1);
+    } else {
+      out = mxnet::TShape(1, 1);
+    }
+    if (in.Size() == 0U) {
```

Comment:
This can be combined with condition `Imperative::Get()->is_np_shape()`. `mxnet::TShape(0, 1)` is the same as `mxnet::TShape(0, -1)`.